### PR TITLE
Added Tests and code for function [correlation_report]

### PR DESCRIPTION
**Pull request by zainnofal**
- Added 5 edge case tests in the test_correlation_report.py as well as docstrings
- Added code for correlation_report.py 
- incorporated changes suggested by LLM assistant 
- removed test_example.py
- changed lined 20 in project.toml that was causing an error when running pytest

Output for pytest -v:
```

============================================ test session starts =============================================
platform darwin -- Python 3.12.11, pytest-8.4.2, pluggy-1.6.0 -- /Users/zainnofal/miniforge3/bin/python
cachedir: .pytest_cache
rootdir: /Users/zainnofal/MDS/Block_4/524/data_fixr
configfile: pyproject.toml
testpaths: tests
plugins: anyio-4.10.0
collected 5 items                                                                                            

tests/unit/test_correlation_report.py::test_correlation_report_three_numeric_columns_values_and_schema PASSED [ 20%]
tests/unit/test_correlation_report.py::test_correlation_report_ignores_non_numeric_columns PASSED      [ 40%]
tests/unit/test_correlation_report.py::test_correlation_report_invalid_method_raises_value_error PASSED [ 60%]
tests/unit/test_correlation_report.py::test_correlation_report_requires_two_numeric_columns PASSED     [ 80%]
tests/unit/test_correlation_report.py::test_correlation_report_non_dataframe_raises PASSED             [100%]

============================================= 5 passed in 0.39s ==============================================


```
Fix for #31, #30, & #41 

Let me know for any changes. Thanks!

### CODE REVIEW

File: src/data_fixr/correlation_report.py [hunk 1]
```diff
@@ -59,4 +61,38 @@ def correlation_report(df, method: str = "pearson"):
     1       age     score         -1.0              1.0
     2    income     score         -1.0              1.0
 
-    """
\ No newline at end of file
+    """
+
+
+    # Validating the Inputss
```
[the remark this answers is not in the document]
Comment (zainnofal) [reply src/data_fixr/correlation_report.py]: Yes fixed it now
